app/src/pages/income.tsx

```diff
@@ -111,36 +148,22 @@ const Income: NextPage = () => {
             />
           </p>
         </fieldset>
-        <ButtonLink href="/choose-clinic" labelKey="continue" />
+        <ButtonLink
+          href={forwardRoute}
+          labelKey={actionButtonLabel}
+          disabled={disabled}
+        />
       </form>
     </>
   )
 }
 
-export const getServerSideProps: GetServerSideProps = async ({
-  locale,
-  req,
-}) => {
-  const prevRouteIndex = req.headers.referer?.lastIndexOf('/')
-  const previousRoute =
-    prevRouteIndex && req.headers.referer?.substring(prevRouteIndex)
-  let returnval: GetServerSidePropsResult<{ [key: string]: object }> = {
+export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
+  return {
     props: {
       ...(await serverSideTranslations(locale || 'en', ['common'])),
     },
   }
-
-  if (!['/eligibility', '/choose-clinic'].includes(previousRoute as string)) {
-    returnval = {
-      ...returnval,
-      redirect: {
-        destination: previousRoute || '/',
-        permanent: false,
-      },
-    }
-  }
-
-  return returnval
```

Comment:
Notes for reviewers: All of this gating logic has been moved into `/src/utils/routing.ts`, specifically `hasRoutingIssues()`, which is called by `_app.tsx` (which you'll see in commit d9e891a380abe832fcabc9826e1ab3dfbb2226f4). See #142 for discussion about `routing.ts`.